reviews/models/mixins.py

```diff
@@ -182,17 +192,83 @@ def resubmission_allowed(self, ev):
+        context['template'] = 'reviews_update_comment'
+        if info.get('notify_comment'):
+            reviews.reviews_email.send(context=context, caller='notify_edit_comment')
+
+    def get_info(self):
+        return {
+            'is_rejected': self.action.to_state == workflow.States.REJECTED.value,
+            'notify_comment': not self.reviewable.provider.reviews_comments_private and self.action.comment,
+            'is_pre_moderation': self.reviewable.provider.reviews_workflow == workflow.Workflows.PRE_MODERATION.value
+        }
+
+    def get_accept_reject_notification(self):
+        info = self.get_info()
+        context = self.get_context()
+        context['template'] = 'reviews_submission_status',
+        context['is_pre_moderation'] = info.get('is_pre_moderation')
+        context['notify_comment'] = info.get('notify_comment')
+        context['is_rejected'] = info.get('is_rejected')
+        reviews.reviews_email.send(context=context, caller='accept_reject_notification')
+
+    def get_context(self):
+        return {
+            'creator': self.reviewable.node.creator,
+            'email_recipients': [contributor._id for contributor in self.reviewable.node.contributors],
+            'settings': settings.DOMAIN,
+            'node': self.reviewable.node,
+            'reviewable_title': self.reviewable.node.title,
+            'reviewable_url': self.reviewable.absolute_url,
+            'provider_comments_private': self.reviewable.provider.reviews_comments_private,
+            'provider': self.reviewable.provider,
+            'provider_url': self.reviewable.provider.external_url if self.reviewable.provider.external_url is not None else settings.DOMAIN + '/preprints' + self.reviewable.provider._id,
+            'provider_contact_email': self.reviewable.provider.email_contact if self.reviewable.provider.email_contact is not None else 'contact@osf.io',
+            'provider_support_email': self.reviewable.provider.email_support if self.reviewable.provider.email_support is not None else 'support@osf.io',
+            'provider_preprint_word': self.reviewable.provider.preprint_word
+        }
+
+@reviews.reviews_email.connect
+def reviews_notification(self, context, caller):
+    timestamp = timezone.now()
+    event_type = utils.find_subscription_type('global_reviews')
+    template = ''.join(context.get('template')) + '.txt.mako'
+    for user_id in context.get('email_recipients'):
+        user = OSFUser.load(user_id)
+        subscriptions = get_user_subscriptions(user, event_type)
+        if user == context.get('creator'):
+            context['is_creator'] = True
+        else:
+            context['is_creator'] = False
+        for notification_type in subscriptions:
+            if ((caller == 'notify_submit' or notification_type != 'none') and not (context['provider_comments_private'] and caller == 'notify_edit_comment') and subscriptions[notification_type] and user_id in subscriptions[notification_type]):
```

Comment:
This is a convoluted lump of conditions... a comment explaining it might be helpful.
Isn't the `comments_private` check handled in `notify_edit_comment`?
Is it right that there's no way for a user to opt out of a `submit` email notification?